### ISSUE TITLE
The Grid filter renders an anchor tag without role=button

### ISSUE THREAD

**Issue by aleksandarevangelatov** [telerik/kendo-ui-core#6997]
### Bug report

The Grid headers renders an `<a>` tag for the Filer button, but lacks a `role="button`. If an element is not a `<button>` it should have an explicitly set `role` to `button`. 

### Reproduction of the problem
1. Inspect the filter in the [Remote Binding Demo](https://demos.telerik.com/kendo-ui/grid/remote-data-binding) The following markup is rendered:
```
<a class="k-grid-filter-menu k-grid-filter" href="#" title="Order Date filter column settings" tabindex="-1"><span class="k-icon k-i-filter"></span></a>
```

### Expected/desired behavior
The Filter element should have a `role="button"

### Environment

* **Kendo UI version:** 2022.2.802
* **Browser:** [all]

**Comment by veselints** [comment]
We have carefully examined the case. While it is true that the element is a `button` and not a `link`, much better solution would be to entirely remove it from the accessibility tree of the document. The element is not focusable and interactive while navigating with keyboard. Hence, the proper approach would be to set its `aria-hidden` attribute to `true`.

**Comment by MilenaCh** [comment]
- [ ] aria-hidden should be added for mvc wrapper also 
Currently filter menu button hasn't aria-hidden(see grid/keyboard-navigation)

**Comment by veselints** [comment]
ping @aleksandarevangelatov
